app/src/main/java/net/gsantner/markor/model/AppSettings.java

```diff
@@ -490,6 +498,11 @@ public boolean getDocumentHighlightState(final String path, final CharSequence c
         return getBool(PREF_PREFIX_HIGHLIGHT_STATE + path, lengthOk && isHighlightingEnabled());
     }
 
+    public boolean getEditorLineNumbersState(final CharSequence chars) {
+        final boolean lengthOk = chars != null && chars.length() < (_isDeviceGoodHardware ? 100000 : 35000);
```

Comment:
I see you used some performance related values here... How well is performance overall, does it greatly impact i.e. scrolling or loading speed? Last time I visited this, it made everything horribly slow

Reply:
Sorry, I didn't do enough testing. It is important but I'm not good at testing, could you have a test for it.

Reply:
Have a moderate to big textfile and scroll a bit. You will immediately notice when it's slow.

Reply:
We should only draw line numbers for the visible lines +- 2 pages.

The highlighter class does this. We can look at it / reuse some code imo.

Reply:
First, the algorithm or method for displaying line numbers is still worth optimizing. Second, the editor's loading strategy for large files may also need to be optimized, such as just loading nearby 1000 lines at a time, which can gracefully load large files. I could try to optimize it if I'm free.

Reply:
@guang-lin Note that Incrementally loading files with Android can be a pain as the underlying file system is unreliable. Our current system (trying to save multiple times etc) is hacky but prevents data loss 99.99% of the time.